lib/_http_client.js

```diff
@@ -65,6 +65,14 @@ function isInvalidPath(s) {
   return false;
 }
 
+function validateHost(host) {
+  if (host !== undefined && host !== null && typeof host !== 'string') {
```

Comment:
It could be more efficient, I have not checked how V8 compiles the two different methods of checking.